be/src/formats/orc/orc_chunk_reader.cpp

```diff
@@ -2281,4 +2282,91 @@ int OrcChunkReader::get_column_id_by_name(const std::string& name) const {
     return -1;
 }
 
-} // namespace starrocks::vectorized
+// ======================================================================================
+
+ORCHdfsFileStream::ORCHdfsFileStream(RandomAccessFile* file, uint64_t length)
+        : _file(std::move(file)), _length(length), _cache_buffer(0), _cache_offset(0), _buffer_stream(_file) {
+    SharedBufferedInputStream::CoalesceOptions options = {.max_dist_size = config::io_coalesce_read_max_distance_size,
+                                                          .max_buffer_size = config::io_coalesce_read_max_buffer_size};
+    _buffer_stream.set_coalesce_options(options);
+}
+
+void ORCHdfsFileStream::prepareCache(orc::InputStream::PrepareCacheScope scope, uint64_t offset, uint64_t length) {
+    const size_t cache_max_size = config::orc_file_cache_max_size;
+    if (length > cache_max_size) return;
+    if (canUseCacheBuffer(offset, length)) return;
+
+    // If this stripe is small, probably other stripes are also small
+    // we combine those reads into one, and try to read several stripes in one shot.
+    if (scope == orc::InputStream::PrepareCacheScope::READ_FULL_STRIPE) {
+        length = std::min(_length - offset, cache_max_size);
+    }
+
+    _cache_buffer.resize(length);
+    _cache_offset = offset;
+    doRead(_cache_buffer.data(), length, offset, true);
```

Comment:
// If this stripe is small, probably other stripes are also small
// we combine those reads into one, and try to read several stripes in one shot.
if there are many small strips, why we alaways read from the beginning?

Reply:
No. read from current position. For example

```
        S0 
 --> S1 [100-200]
        S2 [200-300]
```
when we read s1, we will s1 & s3 in a single shot.